django/forms/models.py

```diff
@@ -573,8 +573,11 @@ class BaseModelFormSet(BaseFormSet):
     unique_fields = set()
 
     def __init__(self, data=None, files=None, auto_id='id_%s', prefix=None,
-                 queryset=None, *, initial=None, **kwargs):
+                 queryset=None, *, edit_only=False, initial=None, **kwargs):
         self.queryset = queryset
+        # if edit_only arg is True in modelformset_factory
+        if not self.edit_only:
+            self.edit_only = edit_only
```

Comment:
I don't think you need to treat this attribute as more special than the others. Something like:

```diff
diff --git a/django/forms/models.py b/django/forms/models.py
index 8853b886ed..cad6de4108 100644
--- a/django/forms/models.py
+++ b/django/forms/models.py
@@ -575,9 +575,6 @@ class BaseModelFormSet(BaseFormSet):
     def __init__(self, data=None, files=None, auto_id='id_%s', prefix=None,
                  queryset=None, *, edit_only=False, initial=None, **kwargs):
         self.queryset = queryset
-        # if edit_only arg is True in modelformset_factory
-        if not self.edit_only:
-            self.edit_only = edit_only
         self.initial_extra = initial
         super().__init__(**{'data': data, 'files': files, 'auto_id': auto_id, 'prefix': prefix, **kwargs})
 
@@ -871,11 +868,11 @@ class BaseModelFormSet(BaseFormSet):
 
 def modelformset_factory(model, form=ModelForm, formfield_callback=None,
                          formset=BaseModelFormSet, extra=1, can_delete=False,
-                         can_order=False, edit_only=False, max_num=None, fields=None, exclude=None,
+                         can_order=False, max_num=None, fields=None, exclude=None,
                          widgets=None, validate_max=False, localized_fields=None,
                          labels=None, help_texts=None, error_messages=None,
                          min_num=None, validate_min=False, field_classes=None,
-                         absolute_max=None, can_delete_extra=True):
+                         absolute_max=None, can_delete_extra=True, edit_only=False):
     """Return a FormSet class for the given Django model class."""
     meta = getattr(form, 'Meta', None)
     if (getattr(meta, 'fields', fields) is None and
@@ -889,13 +886,14 @@ def modelformset_factory(model, form=ModelForm, formfield_callback=None,
                              formfield_callback=formfield_callback,
                              widgets=widgets, localized_fields=localized_fields,
                              labels=labels, help_texts=help_texts,
-                             error_messages=error_messages, field_classes=field_classes)
+                             error_messages=error_messages, field_classes=field_classes,
+                             edit_only=edit_only)
     FormSet = formset_factory(form, formset, extra=extra, min_num=min_num, max_num=max_num,
                               can_order=can_order, can_delete=can_delete,
                               validate_min=validate_min, validate_max=validate_max,
-                              absolute_max=absolute_max, can_delete_extra=can_delete_extra)
+                              absolute_max=absolute_max, can_delete_extra=can_delete_extra,
+                              edit_only=edit_only)
     FormSet.model = model
-    FormSet.edit_only = edit_only
     return FormSet
```

Also, I would put the new arguments at the end of the signature. Since the signature isn't keyword-only, we could break a user's code who was (regrettably) passing arguments without keywords.

Reply:
I can here to suggest something like what Jacob has already said and to correct my point about being included in both places. If it's being passed in via the factory I  don't think it needs to be part of the `__init__` method of `BaseFormSet`.

Reply:
@jacobtylerwalls, I looked at your suggestions and have questions. 
Why you place `edit_only` argument to the `formset_factory`? I see no reason for this. Formset don't creates new objects. 
If David suggest me remove `edit_only` from `__init__` method of  `BaseModelFormset`, maybe, I should put this argument to the `BaseModelForm` and check every form. If the form has `edit_only` as `True`  and creates a new object, we don't save it. What do you think about it?

Reply:
Thanks for following up. In the example in your unit test where the queryset is filtering by a certain author, I have to remember to pass `edit_only=True` every time I have a queryset that I want to create a FormSet for. If my model is so sensitive that I know _every formset I make_ should be edit-only, no matter the queryset, why can't Django let me pass it to the factory and just specify that once?

Or, is there an easier way of accommodating that that I'm just missing? 🤔

Reply:
Thank you for answer. Maybe, I wrote not correct. I understand, why we need `edit_only` in `modelformset_factory`. In my question I mean why you put `edit_only` to the simple `formset_factory` with `modelformset_factory` in your suggestion?

Reply:
Ah, yes I misread you. Sorry. You're right, that part of my suggestion seems unnecessary. I wonder if it was there to get the tests passing. In any case, it would be great to get a version of this pushed to the PR and we can go from there. Thanks again for your work! 👍🏻